src/gt4py/next/program_processors/runners/dace/transformations/map_fusion_extended.py

```diff
@@ -459,6 +460,27 @@ def can_be_applied(
 
         # TODO(phimuell): Implement a check that the array is single use data?
 
+        # Check that the data read by the nested map are not accessed by any other outgoing edge
+        # of the access node.
+        access_node_out_edges = graph.out_edges(self.access_node)
+        access_node_out_edges_data = [edge.data.data for edge in access_node_out_edges]
```

Comment:
This is not guaranteed to work, because in an `AccessNode -> AccessNode` situation it is not clear (nor deterministic) to which of the two AC `edge.data.data` is referring to.
If I understand your correctly then your goal is to find all AC that read _directly_ from `self.access_node` (and by that ignoring all AC that are inside a Map).
In that case you have to do something like this:
```suggestion
        access_node_out_edges_data = [
        oedge.dst.data for oedge in access_node_out_edges if (not oedge.data.is_empty()) and isinstance(oedge, dace_nodes.AccessNode)]  
```
However, the above code handles view in the same way as a normal data container.
This might be a desirable behaviour.